File: front/src/models/Orbat.ts
import { GradeLeger } from "./Grade"
import { PersonnageLeger } from "./Personnage"

export type Orbat =
{
    id: number,
    idParent: number | null,
    titre: string,
    indicatif: string | null,
    frequenceRadio: string | null,
    urlImage: string,
    listeSlot: OrbatSlot[]
}

export type OrbatSlot =
{
    id: number,
    gradeRequis: GradeLeger | null,
    personnage: PersonnageLeger | null,
    role: string,
    ordreAffichage: number,
    estOptionnel: boolean
}

export type OrbatSlotRequete =
{
    /* en cas d'ajout pas d'id */
    id?: number,
    idGradeRequis: number | null,
    idPersonnage: number | null,
    role: string,
    ordreAffichage: number,
    estOptionnel: boolean
}

export type OrbatRequete =
{
    idParent: number | null,
    titre: string,
    indicatif: string | null,
    frequenceRadio: string | null,
    urlImage: string,
    listeSlot: OrbatSlot[]
}

File: back/back/Models/Orbat.cs
using LiteDB;

namespace back.Models;

public sealed class Orbat
{
    [BsonId]
    public int Id { get; set; }

    [BsonRef]
    public Orbat? Parent { get; set; }

    public string? Titre { get; set; }

    public string? Indicatif { get; set; } = null!;

    public string? FrequenceRadio { get; set; }

    public string? NomImage { get; set; }

     public List<OrbatSlot> ListeSlot { get; set; } = [];
}

public sealed class OrbatSlot
{
     public Guid Id { get; set; }

     [BsonRef]
     public Grade? GradeRequis { get; set; }

     [BsonRef]
     public Personnage? Personnage { get; set; }

     public string Role { get; set; } = null!;

     public int OrdreAffichage { get; set; }

     public bool EstOptionnel { get; set; }
}
